### PR TITLE
Rename publish_date to published_date for a geography object

### PR DESCRIPTION
**Pull request by fractalf**
A policy object contains a property published_date which is the exact
same thing as a geography object's publish_date. These should therefore
also have the same name to prevent errors and confusion.

---
name: Default
about: Suggest changes to MDS
title: Rename publish_date to published_date for a geography object

---

# MDS Pull Request

Thank you for your contribution!  Please review our OMF [contributing page](https://github.com/openmobilityfoundation/governance/blob/main/CONTRIBUTING.md) to understand guidelines and policies for participation, and our [Code of Conduct page](https://github.com/openmobilityfoundation/governance/blob/main/CODE_OF_CONDUCT.md).

To avoid complications and help make the Review process as smooth as possible, make sure to:

1. Target [**`dev`**](https://github.com/openmobilityfoundation/mobility-data-specification/tree/dev) branch. Please ensure you are targeting **`dev`**, not **`main`**. 
1. Keep the *"Allow edits from maintainers"* button checked to help us resolve some issues for you.
1. Be ready to resolve any merge conflicts before we approve your Pull Request.
1. Have an up to date profile, per our Github [community profile](https://github.com/openmobilityfoundation/governance/blob/main/CONTRIBUTING.md#community-profile) guildance.

## Explain pull request

A policy object contains a property published_date which is the exact
same thing as a geography object's publish_date. These should therefore
also have the same name to prevent errors and confusion.

## Is this a breaking change

A breaking change would require consumers or implementors of the API to modify their code for it to continue to function (ex: renaming of a required field or the change in data type of an existing field). A non-breaking change would allow existing code to continue to function (ex: addition of an optional field or the creation of a new optional endpoint).

* Yes, breaking

## Impacted Spec

Which spec(s) will this pull request impact?

* `policy`

### CODE REVIEW

**Comment by schnuerle** [comment]
Added some tags and slated for 2.0.0 release since changing the field names. However since this looks like only Geography is being changed, and that hasn't released yet, maybe it can be in 1.1.0.

Also @fractalf make sure to sign the [CLA](https://cla-assistant.io/openmobilityfoundation/mobility-data-specification?pullRequest=597).

File: policy/README.md [hunk 1]
```diff
@@ -289,7 +289,7 @@ An individual `Rule` object is defined by the following fields:
 | `geography_id`   | UUID      | Required   | Unique ID of [Geography](/geography#general-information)                                               |
 | `geography_json`   | UUID      | Required   | The GeoJSON that defines the geographical coordinates.
 | `effective_date`   | [timestamp][ts] | Optional   | `start_date` for first published policy that uses this geo.  Server should set this when policies are published.  This may be used on the client to distinguish between “logical” geographies that have the same name. E.g. if a policy publishes a geography on 5/1/2020, and then another policy is published which references that same geography is published on 4/1/2020, the effective_date will be set to 4/1/2020.
-| `publish_date`   | [timestamp][ts] | Required   | Timestamp that the policy was published, i.e. made immutable                                             |
```
Comment (schnuerle) [inline policy/README.md:292]: This is already 'published_date' in Policy in dev: https://github.com/openmobilityfoundation/mobility-data-specification/blob/dev/policy/README.md#policy

Comment (fractalf) [reply policy/README.md:292]: Sorry, not sure I understand you comment. I changed the part under "Geography" here: https://github.com/openmobilityfoundation/mobility-data-specification/blob/dev/policy/README.md#geography

**Comment by fractalf** [comment]
@schnuerle 
CLA signed

I have limited knownledge to the structure/branches of this project, I just followed the instructions and edited the dev branch.
If the dev branch is targetet for 2.0.0 then I understand.

If you feel this change could be in 1.1.0, I'm happy to make another PR, but the I need to know where to put it? I saw now branch targeting 1.1.0

File: policy/README.md (hunk 1 above)
Comment (schnuerle) [reply policy/README.md:292]: Oh I see you changed the geography field currently within Policy API, not the field in the policy endpoint. Makes sense.

**Comment by schnuerle** [comment]
Thanks for signing.

There is no branch for releases until the OMF makes a release candidate. Everything pretty much gets pointed to 'dev' like you have done already, before going to OMF tech council and board for release approval.  The Milestone tags help categorize features and PRs for releases until they are ready. So you are doing things right!

This whole PR can go into 1.1.0, except for the geography field name change within policy, per [our conversation here](https://github.com/openmobilityfoundation/mobility-data-specification/pull/597#discussion_r521385752), since that would be breaking.  So I went in and edited that away for now.

**Review by schnuerle — APPROVED** [review]
Good for 1.1.0 changes to the new geography endpoint.

**Pull request merged**